node/src/config.rs

```diff
@@ -0,0 +1,154 @@
+use miden_node_block_producer::config::BlockProducerConfig;
+use miden_node_rpc::config::RpcConfig;
+use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::Config;
+use serde::{Deserialize, Serialize};
+
+pub const CONFIG_FILENAME: &str = "miden-node.toml";
+
+// Top-level config
+// ================================================================================================
+
+/// Node top-level configuration.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize, Default)]
+pub struct NodeTopLevelConfig {
+    pub block_producer: BlockProducerConfig,
+    pub rpc: RpcConfig,
+    pub store: StoreConfig,
+}
+
+impl Config for NodeTopLevelConfig {
+    const CONFIG_FILENAME: &'static str = CONFIG_FILENAME;
+}
+
+#[cfg(test)]
+mod tests {
+    use figment::Jail;
+    use miden_node_block_producer::config::{BlockProducerConfig, BlockProducerEndpoint};
+    use miden_node_rpc::config::{RpcConfig, RpcEndpoint};
+    use miden_node_store::config::{StoreConfig, StoreEndpoint};
+    use miden_node_utils::Config;
+
+    use super::{NodeTopLevelConfig, CONFIG_FILENAME};
+
+    #[test]
+    fn test_node_config() {
+        Jail::expect_with(|jail| {
+            jail.create_file(
+                CONFIG_FILENAME,
+                r#"
+                    [block_producer]
+                    store_endpoint = "http://store:8000"
+                    block_producer_endpoint = "http://block_producer:8001"
+
+                    [block_producer.endpoint]
+                    host = "127.0.0.1"
+                    port = 8080
+
+                    [rpc]
+                    store_endpoint = "http://store:8000"
+                    block_producer_endpoint = "http://block_producer:8001"
+
+                    [rpc.endpoint]
+                    host = "127.0.0.1"
+                    port = 8080
+
+                    [store]
+                    sqlite = "local.sqlite3"
+
+                    [store.endpoint]
+                    host = "127.0.0.1"
+                    port = 8080
+                "#,
```

Comment:
In another PR, we should include an example config file, and a test that reads that file and ensures that it is always up-to-date. Otherwise, it's hard to keep track of the structure of the config.

Reply:
> In another PR, we should include an example config file, and a test that reads that file and ensures that it is always up-to-date. Otherwise, it's hard to keep track of the structure of the config.


https://github.com/0xPolygonMiden/miden-node/issues/111

Reply:
> Why do we need block_producer_endpoint?

copy-pasta

Reply:
> Another question: we should be able to collapse properties like so, right?

Yes, it is part of the [TOML language](https://toml.io/en/)